scripts/build-windows.ps1

```diff
@@ -18,28 +14,97 @@ $env:CSC_IDENTITY_AUTO_DISCOVERY = $false
 
 $env:CI = $true
 
-$EXPECTED_NODE_VERSION = (Get-Content .\.nvmrc)
-$ACTUAL_NODE_VERSION = (node -v)
+# -----------------------------------------------------------------------------
+#                  Utility functions
 
-if ( "v$EXPECTED_NODE_VERSION" -ne $ACTUAL_NODE_VERSION)
-{
-  Write-Host "You are not running the expected version of node!"
-  Write-Host "  expected: [v$EXPECTED_NODE_VERSION]"
-  Write-Host "  actual  : [$ACTUAL_NODE_VERSION]"
+Function fail_with_docs {Param ($1)
+  Write-Host "*************** FAILING ***************"
+  Write-Host "$1"
+  Write-Host ""
+  Write-Host "Please read the developer documentation in CONTRIBUTING.md"
   exit 1
 }
 
+Function Test-CommandExists { Param ($command, $1)
+
+  $oldPreference = $ErrorActionPreference
+  $ErrorActionPreference = "stop"
+
+  try {
+    if(Get-Command $command){RETURN}
+  } Catch {
+    fail_with_docs $1
+  }
+  Finally {$ErrorActionPreference=$oldPreference}
+}
+
+# -----------------------------------------------------------------------------
+#                  Checking the developer environment
+# checking for installed programmes
+Test-CommandExists node "Node is not installed"
+Test-CommandExists npm "npm is not installed"
+Test-CommandExists python "Python is not installed"
+# NEEDS proper way to CHECK MSVS Tools
+
+# Checking node.js version
+$EXPECTED_NODE_VERSION = (cat .nvmrc)
+$ACTUAL_NODE_VERSION = (node -v)
+
+if ( "v$EXPECTED_NODE_VERSION" -ne $ACTUAL_NODE_VERSION) {
+  fail_with_docs "You are not running the expected version of node!
+   expected: [v$EXPECTED_NODE_VERSION]
+   actual  : [$ACTUAL_NODE_VERSION]"
+}
+
+# Checking npm version
+$EXPECTED_NPM_VERSION = (Get-Content package.json | ConvertFrom-Json).engines.npm 
+$ACTUAL_NPM_VERSION = (npm -v)
+
+if ( $EXPECTED_NPM_VERSION -ne $ACTUAL_NPM_VERSION) {
+  Write-Host "You are not running the expected version of npm!
```

Comment:
Shouldn't this also refer to `fail_with_docs`?

Reply:
In this case it works just as a warning. This is because we previously test if it is installed (in line 45) and we use the correct version in line 123 (npm i -gf npm@8.7.0).

But if it is more coherent, we can exit the script there as well.


Just noticed that lines 123 and 124 should change as well... They need to be referenced to package.json. I'll commit those changes before you close this PR